src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsbookavideolinkapi/integration/resource/RoomAdministrationIntegrationTest.kt

```diff
@@ -268,6 +273,57 @@ class RoomAdministrationIntegrationTest : IntegrationTestBase() {
     getLocationAttribute(persistedLocationAttributeId).schedule().isEmpty() isBool true
   }
 
+  @Test
+  fun `should amend a scheduled row on a decorated room schedule`() {
+    locationsInsidePrisonApi().stubGetLocationById(norwichLocation)
+
+    webTestClient.createDecoratedRoom(
+      CreateDecoratedRoomRequest(
+        locationUsage = ModelLocationUsage.SCHEDULE,
+        locationStatus = ModelLocationStatus.INACTIVE,
+      ),
+      norwichLocation.toModel(),
+      PROBATION_USER,
+    )
+
+    val schedule = webTestClient.createSchedule(
+      CreateRoomScheduleRequest(
+        locationUsage = ModelLocationScheduleUsage.COURT,
+        startDayOfWeek = 1,
+        endDayOfWeek = 7,
+        startTime = LocalTime.of(9, 0),
+        endTime = LocalTime.of(12, 0),
+        notes = "some notes for the initial schedule",
+      ),
+      norwichLocation.toModel(),
+      COURT_USER,
+    )
+
+    val roomSchedule = webTestClient.amendSchedule(
+      norwichLocation.id,
+      schedule.scheduleId,
+      AmendRoomScheduleRequest(
+        locationUsage = ModelLocationScheduleUsage.PROBATION,
+        startDayOfWeek = 2,
+        endDayOfWeek = 5,
+        startTime = LocalTime.of(9, 0),
+        endTime = LocalTime.of(12, 0),
+        allowedParties = setOf("PROBATION"),
+        notes = "some notes for the amended schedule",
+      ),
+      PROBATION_USER,
+    )
+
+    with(roomSchedule) {
+      locationUsage isEqualTo ModelLocationScheduleUsage.PROBATION
+      startDayOfWeek isEqualTo DayOfWeek.of(2)
```

Comment:
As per out conversation this would probably better as a number and not DayOfWeek on the return type from the API.